code/modules/vending/vending.dm

```diff
@@ -185,6 +185,35 @@
 			boutput(usr, "<span class='alert'>You can't use that as an output target.</span>")
 		return
 
+	MouseDrop_T(mob/target, mob/user)
+		if(!istype(target) || target.buckled || get_dist(user, src) > 1 || get_dist(user, target) > 1 || is_incapacitated(user) || isAI(user) || isAI(target) || isghostcritter(user))
+			return
+
+		actions.start(new/datum/action/bar/icon/shoveMobIntoVendomat(src, target, user), user)
+		sleep(5 SECONDS) // Yeah yeah
+		spitOut(target)
```

Comment:
Shouldn't this be part of the onEnd() of the action? The assumes the action is successful and will result in a `set_loc` on the target.

Reply:
Oops, done!